core/msg.c

```diff
@@ -344,9 +343,9 @@ static int _msg_receive(msg_t *m, int block)
     unsigned state = irq_disable();
 
     DEBUG("_msg_receive: %" PRIkernel_pid ": _msg_receive.\n",
-          sched_active_thread->pid);
+          thread_getpid());
 
-    thread_t *me = (thread_t *)sched_threads[sched_active_pid];
+    thread_t *me = thread_get_active();
```

Comment:
Some thing as above

Reply:
should we make a quick PR just fixing those to directly use `thread_t *me = (thread_t *)sched_active_thread;`? So this PR actually does not change binaries?

I can open the PR.

Reply:
https://github.com/RIOT-OS/RIOT/pull/14717